### PR TITLE
Add Application.Dapr/ConfigurationStores related docs

### PR DESCRIPTION
**Pull request by SoTrx**
## Description
Hey,

This PR adds some documentation on the Dapr Configuration Building Block

## Issue reference

Fixes: radius-project/radius#7889

### CODE REVIEW

File: docs/content/tutorials/tutorial-recipe/index.md [hunk 1]
```diff
@@ -61,14 +61,15 @@ Developers don't need to specify what cloud resources they're using in their app
...
+   default   Applications.Messaging/rabbitMQQueues       bicep                            ghcr.io/radius-project/recipes/local-dev/rabbitmqqueues:latest
+   default   Applications.Dapr/pubSubBrokers             bicep                            ghcr.io/radius-project/recipes/local-dev/pubsubbrokers:latest
+   default   Applications.Dapr/secretStores              bicep                            ghcr.io/radius-project/recipes/local-dev/secretstores:latest
+   default   Applications.Dapr/stateStores               bicep                            ghcr.io/radius-project/recipes/local-dev/statestores:latest
+   default   Applications.Datastores/mongoDatabases      bicep                            ghcr.io/radius-project/recipes/local-dev/mongodatabases:latest
+   default   Applications.Datastores/redisCaches         bicep                            ghcr.io/radius-project/recipes/local-dev/rediscaches:latest
+   default   Applications.Datastores/configurationStores bicep                            ghcr.io/radius-project/recipes/local-dev/configurationStores:latest
```
Comment (lakshmimsft) [inline docs/content/tutorials/tutorial-recipe/index.md:72]: For this entry to be correct and the recipe to exist here, I think we will need an update to the recipes repo: 
https://github.com/radius-project/recipes/blob/main/local-dev/

Comment (SoTrx) [reply docs/content/tutorials/tutorial-recipe/index.md:72]: I'll make a PR for it

File: docs/content/reference/resource-schema/dapr-schema/dapr-configurationStore/snippets/dapr-configuration-recipe.bicep [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+//SAMPLE
+resource pubsub 'Applications.Dapr/configurationStores@2023-10-01-preview' = {
+  name: 'configstore'
+  properties: {
+    environment: environment
+    application: app.id
+    recipe: {
+      // Name a specific recipe to use
+      name: 'azure-redis'
```
Comment (lakshmimsft) [inline docs/content/reference/resource-schema/dapr-schema/dapr-configurationStore/snippets/dapr-configuration-recipe.bicep]: Question: Does this imply that the recipe will include both redis and the configuration for it? Do we want to make that clear/separate?

Comment (SoTrx) [reply docs/content/reference/resource-schema/dapr-schema/dapr-configurationStore/snippets/dapr-configuration-recipe.bicep]: I'm not sure I fully understand your question. This recipe will deploy a [Dapr Configuration Store](https://docs.dapr.io/developing-applications/building-blocks/configuration/howto-manage-configuration/) with Redis as the backend.  The store will hold key-value pairs that applications can use later on. Is there something missing in the recipe ?

Comment (lakshmimsft) [reply docs/content/reference/resource-schema/dapr-schema/dapr-configurationStore/snippets/dapr-configuration-recipe.bicep]: The comment was for the name of the recipe. Should we update it, say, similar to `azure-redis-with-config`.
The current name of recipe `azure-redis` indicates only redis is being deployed.

Comment (SoTrx) [reply docs/content/reference/resource-schema/dapr-schema/dapr-configurationStore/snippets/dapr-configuration-recipe.bicep]: Oh, now I understand ! Sorry, I didn’t catch that the first time. Great idea, i'll rename the recipe

Comment (lakshmimsft) [reply docs/content/reference/resource-schema/dapr-schema/dapr-configurationStore/snippets/dapr-configuration-recipe.bicep]: Hello @SoTrx , just checking on the update.

Comment (SoTrxII) [reply docs/content/reference/resource-schema/dapr-schema/dapr-configurationStore/snippets/dapr-configuration-recipe.bicep]: Done ! Thanks for the reminder

Comment (SoTrx) [reply docs/content/reference/resource-schema/dapr-schema/dapr-configurationStore/snippets/dapr-configuration-recipe.bicep]: Done ! Thanks for the reminder

**Review by lakshmimsft — APPROVED** [review]

**Review by willtsai — APPROVED** [review]

**Pull request merged**